usaspending_api/api_contracts/contracts/v2/agency/toptier_code/budget_function.md

```diff
@@ -0,0 +1,114 @@
+            + Default: `desc`
+            + Members
+                + `desc`
+                + `asc`
+        + `sort` (optional, enum[string])
+            Optional parameter indicating what value results should be sorted by.
+            + Default: `obligated_amount`
+            + Members
+                + `name`
+                + `obligated_amount`
+                + `gross_outlay_amount`
+        + `page` (optional, number)
+            The page number that is currently returned.
+            + Default: 1
+        + `limit` (optional, number)
+            How many results are returned.
+            + Default: 10
+    + Body
+
+            {
+                "fiscal_year": 2018,
+                "order": "asc",
+                "sort": "gross_outlay_amount",
+                "page": 1,
+                "limit": 2
+            }
+
++ Response 200 (application/json)
+    + Attributes
+        + `toptier_code` (required, string)
+        + `fiscal_year` (required, number)
+        + `limit` (required, number)
+        + `page_metadata` (required, PageMetadata, fixed-type)
+            Information used for pagination of results.
+        + `results` (required, array[BudgetFunction], fixed-type)
+        + `messages` (required, array[string], fixed-type)
+            An array of warnings or instructional directives to aid consumers of this endpoint with development and debugging.
+
+    + Body
+
+            {
+                "toptier_code": "086",
+                "fiscal_year": 2018,
+                "limit": 2,
+                "page_metadata": {
+                    "page": 1,
+                    "next": 2,
+                    "previous": null,
+                    "hasNext": true,
+                    "hasPrevious": false
+                },
+                "results": [
+                    {
+                        "name": "Health",
+                        "children": [
+                            {
+                                "name": "Health care services",
+                                "obligated_amount": 4982.19,
+                                "gross_outlay_amount": 4982.19
+                            }
+                        ],
+                        "obligated_amount": 4982.19,
+                        "gross_outlay_amount": 4982.19
+                    }
+                ],
+                "messages": []
+            }
+
+# Data Structures
+
+## BudgetFunction (object)
++ `name` (required, string)
++ `obligated_amount` (required, number)
++ `gross_outlay_amount` (required, number)
++ `children` (required, array[BudgetSubFunction], fixed-type)
+
+## BudgetSubFunction (object
++ `name` (required, string)
++ `obligated_amount` (required, number)
++ `gross_outlay_amount` (required, number)
+
+## PageMetadata (object)
++ `page` (required, number)
++ `next` (optional, number)
++ `previous` (optional, number)
```

Comment:
Some of these are nullable... I leave it to the author to figure out which.  `next` and `previous` for sure.

Reply:
Looks like just `next` and `previous`.  Would you mind updating `budget_function` and `program_activity` as well?  Looks like they're also wrong.

Reply:
gotcha